cmd/server/main.go

```diff
@@ -94,6 +94,15 @@ func main() {
 	}()
 
 	go func() {
+		// Server should unlink the socket file prior to binding it in case the socket isn't cleaned up gracefully.
+		// This can happen if the application is killed by SIGKILL or SIGSTOP, i.e. kill -9 or docker kill by default.
+		zap.L().Info("unlinking ", zap.String("port", *addr))
+		err = syscall.Unlink(*addr)
+		if err != nil {
+			if err.Error() != "no such file or directory" {
+				zap.L().Warn("Failed to unlink unix socket. %v", zap.Error(err))
```

Comment:
You'll probably remove this line and replace it with a `Fatal` when you can't remove the socket, but the `%v` is unnecessary here as `zap.Error()` adds an `{"error": "<message>"}` field to your log line

```go
zap.L().Fatal("Failed to remove existing socket", zap.String("address", *addr), zap.Error(err))
```